packages/expo-updates/android/src/main/java/expo/modules/updates/loader/RemoteLoader.java

```diff
@@ -75,10 +77,12 @@ public void onFailure(String message, Exception e) {
 
       @Override
       public void onSuccess(Manifest manifest) {
+        mManifest = manifest;
         if (mCallback.onManifestLoaded(manifest)) {
           processManifest(manifest);
         } else {
-          mCallback.onSuccess(null);
+          mUpdateEntity = null;
+          finishWithSuccess();
```

Comment:
So in the PR summary you wrote:

>If the manifest loads successfully, or is ignored because the client has the same or a newer update already, the parsed data is saved to disk using the json_data SQLite table.

does this mean that the header data is used to select the manifest the same time it is received or is it saved and applied for the next time the app is opened?

Reply:
Thanks for the question -- that behavior isn't actually implemented in this PR (reading & applying the saved filters will be the **next** PR/set of PRs) but this is critical to clarify regardless.

I am planning on essentially the first option -- when we get a new manifest with new header data, we'll persist that data right away after downloading the update successfully (or determining we already have the update), and then immediately read from disk and use that new header data to select the new update to launch. (That said, this may **all** be happening in the background after launching a previously downloaded update, if you have asynchronous automatic updates configured.)

Reply:
Let me know if you think that makes sense or if you foresee any issues/weird edge cases that might happen there. This is definitely the gnarliest part of this whole feature.

Reply:
OK thanks for clarifying that.

Seems like the best option as well. (Assuming writing to the db doesn't cause noticeable latency).

Reply:
That's a good point -- maybe we can do some profiling once the entire feature is landed & works end to end, and then figure out if we need to add any optimizations. For now just wanted to focus on correctness rather than speed.